src/Aspire.Hosting.Orleans/OrleansServiceExtensions.cs

```diff
@@ -348,25 +380,40 @@ public static class OrleansServiceExtensions
             provider.ConfigureResource(builder, $"BroadcastChannel__{name}");
         }
 
-        if (!string.IsNullOrWhiteSpace(res.ClusterId))
+        builder.WithEnvironment(context =>
         {
-            builder.WithEnvironment("Orleans__ClusterId", res.ClusterId);
-        }
-
-        if (!string.IsNullOrWhiteSpace(res.ServiceId))
-        {
-            builder.WithEnvironment("Orleans__ServiceId", res.ServiceId);
-        }
+            context.EnvironmentVariables["Orleans__ClusterId"] = res.ClusterId;
+            context.EnvironmentVariables["Orleans__ServiceId"] = res.ServiceId;
 
-        // Set silo-to-silo and client-to-gateway ports
-        // Note that we sets these ports even when adding a reference to a client.
-        builder.WithEndpoint(scheme: "tcp", name: "orleans-silo", env: "Orleans__Endpoints__SiloPort");
-        builder.WithEndpoint(scheme: "tcp", name: "orleans-gateway", env: "Orleans__Endpoints__GatewayPort");
+            // Enable distributed tracing by default
+            if (res.EnableDistributedTracing != false)
+            {
+                context.EnvironmentVariables["Orleans__EnableDistributedTracing"] = "true";
+            }
+        });
 
-        // Enable distributed tracing by default
-        if (res.EnableDistributedTracing != false)
+        if (isSilo)
         {
-            builder.WithEnvironment("Orleans__EnableDistributedTracing", "true");
+            if (res.Reminders is { } reminders)
+            {
+                reminders.ConfigureResource(builder, "Reminders");
+            }
+
+            foreach (var (name, provider) in res.GrainStorage)
+            {
+                provider.ConfigureResource(builder, $"GrainStorage__{name}");
+            }
+
+            foreach (var (name, provider) in res.GrainDirectory)
+            {
+                provider.ConfigureResource(builder, $"GrainDirectory__{name}");
+            }
+
+            // Set silo-to-silo and client-to-gateway ports
+            // NOTE: These endpoints are specified as proxied even though we never expect any connections via the proxy, and that would be invalid anyway.
+            // this is a workaround for current Aspire/DCP behavior.
+            builder.WithEndpoint(scheme: "tcp", name: "orleans-silo", env: "Orleans__Endpoints__SiloPort", isProxied: true, isExternal: false);
+            builder.WithEndpoint(scheme: "tcp", name: "orleans-gateway", env: "Orleans__Endpoints__GatewayPort", isProxied: true, isExternal: false);
```

Comment:
I'm a bit confused by this comment.

1. Isn't "isProxied" defaulted to `true`?
2. What exactly is this working around?

Reply:
IsProxied is defaulted to true. In this case, we don't actually want a proxy, but we need to make sure IsProxied is set to true, because otherwise DCP will barf. It's a limitation. The proxy ends up being unused by the app (they connect directly to each other), so it's benign but worth mentioning in a comment and being explicit about the current behavior.